[WebGPU] Crash in ConstantValue::integerValue() because the activeType is float

mwyrzykowski opened the pull request:
#### 81a7c89e229f5fd8c595c90f11054bd3b35c1d9c
<pre>
[WebGPU] Crash in ConstantValue::integerValue() because the activeType is float
<a href="https://bugs.webkit.org/show_bug.cgi?id=265894">https://bugs.webkit.org/show_bug.cgi?id=265894</a>
&lt;radar://119206116&gt;

Reviewed by Tadeu Zagallo.

The wgslConstantValues should use the source name, not the mangledName.

We see the crash because the override was not found, so it returned an
empty value with default value of float.

* Source/WebGPU/WebGPU/Pipeline.mm:
(WebGPU::createConstantValues):

Canonical link: <a href="https://commits.webkit.org/271743@main">https://commits.webkit.org/271743@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4af6e3136cc3ed16563c119688807611553a3d4e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29288 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7958 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/30625 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/31866 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/26621 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10098 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5259 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/26622 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/29561 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/6615 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25077 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5695 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/5836 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33206 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26745 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/26555 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32054 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5803 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3982 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29839 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7490 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7006 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6317 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6327 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

djg commented on Source/WebGPU/WGSL/ConstantValue.h:
```diff
@@ -148,6 +148,10 @@ struct ConstantValue : BaseValue {
         if (auto* abstractInt = std::get_if<int64_t>(this))
             return *abstractInt;
+        if (auto* abstractFloat = std::get_if<float>(this)) {
+            RELEASE_ASSERT(*abstractFloat == 0.f);
```
@tadeuzagallo Should this follow the C rules of converting a float to int by truncation?

The spec (https://www.w3.org/TR/WGSL/#conversion-rank) appears to ban the automatic conversion of float to int.

tadeuzagallo replied:
I don't think this is correct. It seems like we have a mismatch between the constant and the type, which is incorrect. We're probably not converting the override initial value, which we need to.

mwyrzykowski replied:
I agree, I will investigate a bit further to see what is occurring here.

mwyrzykowski replied:
@djg @tadeuzagallo ah so we were using mangledName where we should not have been. 

PR is updated when you have a minute

mwyrzykowski committed:
[WebGPU] Crash in ConstantValue::integerValue() because the activeType is float
https://bugs.webkit.org/show_bug.cgi?id=265894
<radar://119206116>

Reviewed by Tadeu Zagallo.

The wgslConstantValues should use the source name, not the mangledName.

We see the crash because the override was not found, so it returned an
empty value with default value of float.

* Source/WebGPU/WebGPU/Pipeline.mm:
(WebGPU::createConstantValues):

Canonical link: https://commits.webkit.org/271743@main
